applications/feedback_sentiment_analyzer/SPECIFICATION.md

```diff
@@ -0,0 +1,224 @@
+| `TranslateText` complete.    | [SynthesizeAudio](#SynthesizeAudio)  | Synthesize   human-like audio from text. |
+
+#### Diagram
+These functions appear by name in the multi-state ]workflow depicted below. 
+
+Also depicted is where the workflow forks at `ContinueIfPositive` based whether sentiment analyzed is `POSITIVE` or `NEGATIVE`. 
+
+When ran successfully, the state machine  adds a new item to a DynamDB table containing data gathered during the workflow.
+
+![FSA - l'hotellerie.png](..%2F..%2F..%2F..%2F..%2FDownloads%2FFSA%20-%20l%27hotellerie.png)
+
+Logging appears in the AWS Console under Step Functions and (for Lambda functions only) CloudWatch.
+
+## Function inputs and outputs
+See below for the required inputs and outputs of each Lambda function.
+
+### ExtractText
+
+#### **Input**
+Function will use data available on the [S3 event object](https://docs.aws.amazon.com/AmazonS3/latest/userguide/EventBridge.html). 
+
+For example:
+```json
+{
+  "bucket": "my-s3-bucket",
+  "region": "us-east-1",
+  "object": "foo/bar.png"
+}
+```
+#### **Output**
+Function will return a string representing the text extracted.
+
+For example:
+```json
+THIS HOTEL WAS GREAT
+```
+
+### AnalyzeSentiment
+#### **Input**
+Function will use data available on the [Lambda event object](https://docs.aws.amazon.com/lambda/latest/dg/gettingstarted-concepts.html#gettingstarted-concepts-event). 
+
+For example:
+```json
+{
+    "source_text": "THIS HOTEL WAS GREAT",
+    "region": "us-east-1"
+ }
+```
+#### **Output**
+Function will return a string representing the text extracted.
+
+For example:
+```json
+{
+   "sentiment": 'POSITIVE',
+   "language_code": 'en'
+}
+```
+### TranslateText
+#### **Input**
+Function will use data available on the [Lambda event object](https://docs.aws.amazon.com/lambda/latest/dg/gettingstarted-concepts.html#gettingstarted-concepts-event).
+
+For example:
+```json
+{
+    "source_language_code": "en"
+    "region": "us-east-1",
+    "extracted_text": "THIS HOTEL WAS GREAT"
+}
+```
+#### **Output**
+Function will return a string representing the text translated.
+
+For example:
+```json
+CET HOTEL ÉTAIT RAVISSANT
+```
+
+### SynthesizeAudio
+#### **Input**
+Function will use data available on the [Lambda event object](https://docs.aws.amazon.com/lambda/latest/dg/gettingstarted-concepts.html#gettingstarted-concepts-event).
+
+For example:
+```json
+{
+    "bucket": "my-s3-bucket",
+    "translated_text": "CET HOTEL ÉTAIT RAVISSANT",
+    "region": "us-east-1",
+    "object": "comment.png"
+ }
+```
+#### **Output**
+Function will return a string representing the key of the newly-synthesized audio file.
+
+For example:
+```json
+my-s3-bucket/audio.mp3
+```
+---
+
+## GetFeedback Lambda function
+To display data on previously-uploaded images, the frontend invokes a Lambda function which fetches items from a [DynamoDB table](#DynamoDB).
```

Comment:
We should note that this function is hardcoded as part of the CDK deployment and does not need to be implemented by specific languages.

Reply:
Add explanation like the one proposed above